### PR TITLE
Refactor file structure

### PR DESCRIPTION
**Pull request by mackenziequinnjetton**
https://trello.com/c/TOIXzZiP/5-tic-tac-toe-refactor-file-structure

## Added
* gitignore
* gemfiles
## Changed
* introduced lib folder for source code
* moved test code into a spec folder

pair: @Aoxian

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require_relative 'board_row'
+require_relative '../lib/board_row'
```
Comment (Aoxian) [inline lib/board.rb:1]: While we were taking small steps moving the files around, we introduced the relative paths.  As the test still passed, I believe we overlooked removing those paths once they became redundant.  So, for this example, the `board_row.rb` file is in the same relative folder. Therefore you should be able to clean this up further.

```suggestion
require_relative 'board_row'
```

Comment (Aoxian) [reply lib/board.rb:1]: This is just one instance, I would suggest checking all the files and cleanup appropriately.

Comment (mackenziequinnjetton) [reply lib/board.rb:1]: Went through every file and fixed.

**Review by Aoxian — APPROVED** [review]

**Comment by kevbuchanan** [comment]
A few notes:

- A `README.md` file would be nice to have, in order to document what the project is and how to run basic tasks like starting the CLI, or running tests.
- A `Rakefile` with a `test` [task](https://docs.ruby-lang.org/en/2.1.0/Rake/TestTask.html) would be nice to have in order to run all tests. Then add instructions for the `rake test` command to the `README`.
- You'll also typically want to use the Ruby `LOAD_PATH` to require code, rather than using `require_relative`. That can be configured by [adding the lib directory to your load path](https://thoughtbot.com/blog/following-the-path) in your tests. the `Rake::TestTask` will do this for you when you configure the [libs](https://docs.ruby-lang.org/en/2.1.0/Rake/TestTask.html) attribute.

**Comment by mackenziequinnjetton** [comment]
> A few notes:
> 
> * A `README.md` file would be nice to have, in order to document what the project is and how to run basic tasks like starting the CLI, or running tests.
> * A `Rakefile` with a `test` [task](https://docs.ruby-lang.org/en/2.1.0/Rake/TestTask.html) would be nice to have in order to run all tests. Then add instructions for the `rake test` command to the `README`.
> * You'll also typically want to use the Ruby `LOAD_PATH` to require code, rather than using `require_relative`. That can be configured by [adding the lib directory to your load path](https://thoughtbot.com/blog/following-the-path) in your tests. the `Rake::TestTask` will do this for you when you configure the [libs](https://docs.ruby-lang.org/en/2.1.0/Rake/TestTask.html) attribute.

Alright. I'm aware of what `README.md` files are, _I think,_ but I do not know how to write one. I do not know what `rake` or `Rakefile` are or how to use them. And I do not know what `LOAD_PATH` is; I read the article you linked, but I do not understand what I am reading. In summary, either I need more guidance on these topics, or we can table them and I can move ahead for now, with us circling back to them another day.

**Comment by kevbuchanan** [comment]
`README.md` is a document that explains anything someone might need to know about the project, i.e. how to build it, run it, and any dependencies. `.md` is the file extension for a [markdown](https://www.markdownguide.org/basic-syntax/) document, which is a markup language commonly used for project documentation.

[rake](https://github.com/ruby/rake) is a task runner for Ruby. You'll create a `Rakefile` which defines the tasks according to the rake syntax.

In Ruby, the load path is a global variable (`$LOAD_PATH`) that the interpreter uses to load files through `require` statements. https://webapps-for-beginners.rubymonstas.org/libraries/load_path.html

**Comment by mackenziequinnjetton** [comment]
> `README.md` is a document that explains anything someone might need to know about the project, i.e. how to build it, run it, and any dependencies. `.md` is the file extension for a [markdown](https://www.markdownguide.org/basic-syntax/) document, which is a markup language commonly used for project documentation. Here's an example `README`:
> 
> * Raw file: https://raw.githubusercontent.com/qcam/3llo/master/README.md
> * Rendered markdown (handled by GitHub): https://github.com/qcam/3llo/blob/master/README.md
> 
> [rake](https://github.com/ruby/rake) is a task runner for Ruby. You'll create a `Rakefile` which defines the tasks according to the rake syntax.
> 
> In Ruby, the load path is a global variable (`$LOAD_PATH`) that the interpreter uses to load files through `require` statements. https://webapps-for-beginners.rubymonstas.org/libraries/load_path.html

Understood. I will get these all implemented and submit them as a separate pull request.

**Pull request merged**